luaui/Widgets/gui_chat.lua

```diff
@@ -561,9 +561,21 @@ end
 local energyText = Spring.I18N('ui.topbar.resources.energy'):lower()
 local metalText = Spring.I18N('ui.topbar.resources.metal'):lower()
 
+local function filterBadword(text)
+	badWords = {"retard"}
```

Comment:
This is a global variable (doesn't have `local` in front of it, and it's set every time the function runs. Better to add `local badWords = {...}` outside the function, like `metalText`

Reply:
Please avoid using globals. Store in WG or GG if it must be available to all.